pkg/server/sandbox_run.go

```diff
@@ -452,13 +453,42 @@ func (c *criService) generateSandboxContainerSpec(id string, config *runtime.Pod
 	}
 	g.SetProcessOOMScoreAdj(adj)
 
+	for passthroughAnnotationKey, passthroughAnnotationvalue := range c.getPassthroughAnnotations(config.Annotations) {
+		g.AddAnnotation(passthroughAnnotationKey, passthroughAnnotationvalue)
+	}
+
 	g.AddAnnotation(annotations.ContainerType, annotations.ContainerTypeSandbox)
 	g.AddAnnotation(annotations.SandboxID, id)
 	g.AddAnnotation(annotations.SandboxLogDir, config.GetLogDirectory())
 
 	return g.Config, nil
 }
 
+func (c *criService) getPassthroughAnnotations(sandboxAnnotations map[string]string) (passthroughAnnotations map[string]string) {
+	passthroughAnnotations = make(map[string]string)
+
+	for podAnnoationKey, podAnnoationValue := range sandboxAnnotations {
+		for runtimeKey := range c.config.Runtimes {
```

Comment:
within RunPodSandbox() call r.GetRuntimeHandler() to get the runtimeKey you need... pass that to this getPassthroughAnnotations() helper.. use the passed in key as the runtimeKey.. 

check if there is a runtime in c.config.Runtimes[runtimeKey] before you start the loop.. at the top of the function and return nil if there are no matching runtimes...

hope this helps

Reply:
Thanks @mikebrow . That really helped. Turns out `ociRuntime` is already [fetched](https://github.com/harche/cri/blob/passthrough_annotations/pkg/server/sandbox_run.go#L101) within `RunPodSandbox()` so I just have to [pass](https://github.com/harche/cri/blob/passthrough_annotations/pkg/server/sandbox_run.go#L151) the runtime pod annotations defined for that exact runtime all the way down till `getPassthroughAnnotations()`.

Thanks for the pointer. I have updated my PR.